hetmech/dwpc.py

```diff
@@ -204,15 +207,176 @@ def dwpc_long_repeat(graph, metapath, damping=0.5):
     raise NotImplementedError("See PR #59")
 
 
+def categorize(metapath):
+    """
+    Returns the classification of a given metapath as one of
+    a set of metapath types which we approach differently.
+    Parameters
+    ----------
+    metapath : hetio.hetnet.MetaPath
+
+    Returns
+    -------
+    classification : string
+        One of ['no_repeats', 'disjoint', 'short_repeat',
+                'long_repeat', 'BAAB', 'BABA', 'other']
+    Examples
+    --------
+    GbCtDlA -> 'no_repeats'
+    GiGiG   -> 'short_repeat'
+    GiGiGcG -> 'long_repeat'
+    GiGcGiG -> 'long_repeat'
+    GiGbCrC -> 'disjoint'
+    GbCbGbC -> 'BABA'
+    GbCrCbG -> 'BAAB'
+    DaGiGbCrC -> 'disjoint'
+    GiGaDpCrC -> 'disjoint'
+    GiGbCrCpDrD -> 'disjoint'
+    GbCpDaGbCpD -> NotImplementedError
+    GbCrCrCrCrCbG -> NotImplementedError
+    """
+    metanodes = list(metapath.get_nodes())
+    repeated_nodes = {v for i, v in enumerate(metanodes) if
```

Comment:
Now that you have the counter, it may be more straightforward to just do:

```python
{metanode for metanode, count in freq.items() if count > 1}
```